packages/graph/src/Graph.ts

```diff
@@ -448,10 +448,42 @@ export class Graph extends SVGZoomWidget {
         vertexElements.order();
     }
 
+    setTooltipType() {
+        if (this.useDetailedTooltip()) {
+            const context = this;
+            this.tooltipHTML(function (d: any) {
+                if (d instanceof Subgraph) {
+                    return `<p style="text-align:center">${d.title().replace("\n", "<br>")}</p>`;
+                } else if (d instanceof Vertex) {
+                    return context.tooltipKeyValueFormat({
+                        label: d.text().replace("\n", "<br>"),
+                        connections: context.data().edges.filter(e => e.targetVertex().id() === d.id() || e.sourceVertex().id() === d.id()).length
+                    });
+                } else if (d instanceof Edge) {
+                    return `<p style="text-align:center">${d.text().replace("\n", "<br>")}</p>`;
+                }
+                return "";
+            });
+        } else {
+            this.tooltipHTML(function (d: any) {
+                if (d instanceof Subgraph) {
+                    return `<p style="text-align:center">${d.title().replace("\n", "<br>")}</p>`;
+                } else if (d instanceof Vertex) {
+                    return `<p style="text-align:center">${d.text().replace("\n", "<br>")}</p>`;
+                } else if (d instanceof Edge) {
+                    return `<p style="text-align:center">${d.text().replace("\n", "<br>")}</p>`;
+                }
+                return "";
+            });
+        }
+    }
+
     update(domNode, element) {
         super.update(domNode, element);
         this.tooltip.hide();
 
+        this.setTooltipType();
```

Comment:
Can't do this here as the user may have already provided their own tooltip function.